pac4j-saml/src/main/java/org/pac4j/saml/crypto/KeyStoreCredentialProvider.java

```diff
@@ -114,12 +148,17 @@ private KeyStore loadKeyStore(final InputStream inputStream, final String storeP
         }
     }
 
-    private String getPrivateKeyAlias(final KeyStore keyStore) {
+	private String getPrivateKeyAlias(final KeyStore keyStore, final String keyStoreAlias) {
         try {
             final Enumeration<String> aliases = keyStore.aliases();
-            if (aliases.hasMoreElements()) {
-                return aliases.nextElement();
-            }
+			while (aliases.hasMoreElements()) {
+	            String currentAlias = aliases.nextElement();
+				if (keyStoreAlias == null) {
+					return currentAlias;
+				} else if (currentAlias.equals(keyStoreAlias)) {
```

Comment:
In both cases, you return the `currentAlias`, you could have one condition with `||`.